diff --git a/views_plugin_pager_infinite_scroll.inc b/views_plugin_pager_infinite_scroll.inc
index 6e09c24..24853ac 100644
--- a/views_plugin_pager_infinite_scroll.inc
+++ b/views_plugin_pager_infinite_scroll.inc
@@ -62,28 +62,25 @@ class views_plugin_pager_infinite_scroll extends views_plugin_pager_full {
     $items_selector = '';
     $img_path = $base_url . '/' . drupal_get_path('module', 'views_infinite_scroll') . '/images/ajax-loader.gif';
     switch ($this->view->plugin_name) {
-      case 'default':
-        $content_selector = 'div.view-content';
-        $items_selector = '.views-row';
-        break;
       case 'grid':
         $content_selector = 'div.view-content > table > tbody';
         $items_selector = 'tr';
         break;
       case 'list':
         if (array_key_exists('wrapper_class', $style_options) && !empty($style_options['wrapper_class'])) {
-          $wrapper_class = '.' . $style_options['wrapper_class'];
+          $content_selector = 'div.view-content > .' . $style_options['wrapper_class'] . ' > *';
+        } else {
+          $content_selector = 'div.view-content > *';
         }
-        else {
-          $wrapper_class = '.item-list';
-        }
-        $content_selector = 'div.view-content>' . $wrapper_class . ' > *';
         $items_selector = '.views-row';
         break;
       case 'table':
         $content_selector = 'div.view-content > table > tbody';
         $items_selector = 'tr';
         break;
+      default:
+        $content_selector = 'div.view-content';
+        $items_selector = '.views-row';
     }
     $pager_theme = views_theme_functions('views_infinite_scroll_pager', $this->view, $this->display);
     return theme($pager_theme, array(
